libdevice/nativecpu_utils.cpp

```diff
@@ -29,6 +29,11 @@ using __nativecpu_state = native_cpu::state;
 #define DEVICE_EXTERNAL_C DEVICE_EXTERN_C __attribute__((always_inline))
 #define DEVICE_EXTERNAL SYCL_EXTERNAL __attribute__((always_inline))
 
+// Several functions are used implicitly by WorkItemLoopsPass and
+// PrepareSYCLNativeCPUPass and need to be marked as used to prevent them being
+// removed early.
+#define USED __attribute__((used))
```

Comment:
Nit; "USED" seems a little general, but I doubt that it will ever conflict with anything, so it's probably fine.

Reply:
True, if this were used in a header file it would be an issue. In a source file, I can't imagine a problem coming up either.